drivers/syscon/CMakeLists.txt

```diff
@@ -1,4 +1,4 @@
 # SPDX-License-Identifier: Apache-2.0
 
 zephyr_library()
-zephyr_library_sources_ifdef(CONFIG_SYSCON          syscon.c)
+zephyr_library_sources_ifdef(CONFIG_SYSCON_GENERIC          syscon.c)
```

Comment:
Leave `CONFIG_SYSCON` for now. To be decided if there will be ever other types.

Reply:
Done